packages/react-components/src/components/chart/hooks/useViewportToMS.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+import { isDurationViewport, convertViewportToMs } from '../utils/getInfo';
+import { Viewport } from '@iot-app-kit/core';
+import { LIVE_MODE_REFRESH_RATE_MS } from '../eChartsConstants';
+
+export const useViewportToMS = (viewport?: Viewport) => {
+  const [inMS, setInMS] = useState(convertViewportToMs(viewport));
+
+  const viewportString = JSON.stringify(viewport);
+  useEffect(() => {
+    let interval: NodeJS.Timer;
+    if (viewport && isDurationViewport(viewport)) {
+      interval = setInterval(() => {
+        setInMS(convertViewportToMs(viewport));
+      }, LIVE_MODE_REFRESH_RATE_MS);
+    } else {
+      setInMS(convertViewportToMs(viewport));
```

Comment:
what does milliseconds represent in an absolute viewport? the difference between the start and end?

Reply:
just converted to ms, initial, end and delta between  them.. all in ms